metricsbp/config.go

```diff
@@ -53,6 +53,12 @@ type Config struct {
 	//
 	// Optional, defaults to false.
 	RunSysStats bool `yaml:"runSysStats"`
+
+	// ReportServerConnectionCount indicates that you want to publish
+	// a counter for the number of clients connected to the server
+	//
+	// Optional, defaults to false.
+	ReportServerConnectionCount bool `yaml:"reportServerConnectionCount"`
```

Comment:
Is there a way to set the server config from the application yaml?

Reply:
Is that desireable? Metrics are not usually the domain of configuration

Reply:
we can also just remove this configuration and just report it always. an additional counter/gauge adds negligible overhead to a go service.

Reply:
Let me know if I should update it to always enabled

Reply:
@kylelemons what do you think to remove this configuration and just always on?

Reply:
The metric is negligible but the additional thunk call might not be... though it may be dwarfed by thrift overhead.

I don't have enough experience to know, but if both of you agree it's ok to enable by default then it probably is

Reply:
which is why I prefer to just move it to `thriftbp.ServerConfig` instead of default on :)

in the current cookiecutter we always need to pass in `thriftbp.ServerConfig` to `thriftbp.NewBaseplateServer` manually, so adding a config there manually to make it opt-in is not something too out of the scope.

If you want it from yaml, you can just make `thriftbp.ServerConfig` part of your `Config` struct and override the `Processor` field (as that cannot be deserialized from yaml anyways).